src/pages/Balanceboard.tsx

```diff
@@ -0,0 +1,130 @@
+import { useState, useEffect } from 'react'
+import { fetcher } from '../utils'
+import EChartsReact from 'echarts-for-react'
+
+export default function Balanceboard() {
+    const [data, setData] = useState([])
+    const [condition, setCondition] = useState({ method: 'year' })
+    const [xAxis, setxAxis] = useState<string[]>([])
+    const [pcr, setPcr] = useState<number[]>([])
+    const [rtkag, setRtkag] = useState<number[]>([])
+    const [option, setOption] = useState<object>({
+        xAxis: {
+            type: 'category',
+            data: xAxis
+        },
+        yAxis: {
+            type: 'value'
+        },
+        series: [
+            {
+                data: pcr,
+                type: 'line'
+            }
+        ]
+    })
+    const getData = async () => {
+        let OriginalData = await fetcher("epidemic/tests_malaysia.csv")
+        console.log(OriginalData);
+        const xArr: string[] = []
+        const pcr: number[] = []
+        const rtkag: number[] = []
+        OriginalData = OriginalData.filter((item: any) => {
+            const itemDate: string = item.date
+            const itemDateArr = itemDate.split('-')
+            const itemYear: number = Number(itemDateArr[0])
+            const itemMonth: number = Number(itemDateArr[1])
+            const itemDay: number = Number(itemDateArr[2])
+
+            const currentDate = new Date()
+            const currentYear = currentDate.getFullYear()
+            const currentMonth: number = currentDate.getMonth() + 1
+
+            switch (condition.method) {
+                case 'year':
+                    if (itemYear === currentYear) {
+                        xArr.push(itemDate)
+                        pcr.push(item['pcr'])
+                        rtkag.push(item['rtk-ag'])
+                    }
+                    return itemYear === currentYear
+                case 'month':
+                    if (itemYear === currentYear && itemMonth === currentMonth) {
+                        xArr.push(itemDate)
+                        pcr.push(item['pcr'])
+                        rtkag.push(item['rtk-ag'])
+                    }
+                    return itemYear === currentYear && itemMonth === currentMonth
+                default:
+                    break;
+            }
+        })
+        setxAxis(xArr)
+        setPcr(pcr)
+        setRtkag(rtkag)
```

Comment:
Hi Lei, you could put this in the loader function and return xArr, pcr, rtkag as an object like this `{ xArr, pcr, rtkag }`